File: frontend/src/app/shared/interview/intake-dto.ts
import { Intake } from "./intake";

export class IntakeDTO {
    drugId?: number;
    startDate?: Date;
    endDate?: Date;
    prescribed: boolean = false;
    regularly: boolean = false;
    dosePerDay?: number;
    intakeInterval?: string;
    doseUnit?: string;
    takenToday: boolean = false;
    note?: string;
    sourceOfDrugData?: string;

    public static fromIntake(intake: Intake): IntakeDTO {
        var dto: any = Object.assign({}, intake);
        delete dto.drug;
        dto.drugId = intake.drug?.pharmaceuticalCentralNumber;
        return dto as IntakeDTO;
    }
}

File: backend/src/main/java/de/dzd/idom/intake/IntakeDTO.java
package de.dzd.idom.intake;

import lombok.Data;

import java.time.LocalDate;

@Data
public class IntakeDTO {
    private LocalDate startDate;
    private LocalDate endDate;
    private boolean isPrescribed;
    private boolean isRegularly;
    private boolean isTakenToday;
    private String note;
    private long drugId;

    private int dosePerDay;
    private String intakeInterval;
    private String doseUnit;
}
